### ISSUE TITLE
BLEU-4 is only about 0.15 which is dramatically different with the score 31.4

### ISSUE THREAD

**Issue by bozhenhhu** [eric-xw/Video-guided-Machine-Translation#10]
Dear Eric, 
    Thank you for your great work. I have learnt a lot for it .
    I use the original codes to trian the  baseline VMT model. The training information is as follows. And I tested the submission.json, I got Corpus Bleu-4: 0.284365163774
 B1: 0.576045493433
 B2: 0.340936309609
 B3: 0.22165432701
 B4: 0.150210162101
Why the B4 score is so different with your score 31.4,  I see the results in https://competitions.codalab.org/competitions/24384#results    are also different with 31.4. **What cause this difference? Maybe you have used another method to calculate the BLEU?**

Training model: s2s_en2zh
Writing vocab of size 10523 to data/train_vocab_en.txt
Writing vocab of size 2907 to data/train_vocab_zh.txt
{'TRAIN_VOCAB_EN': 'data/train_vocab_en.txt', 'TRAIN_VOCAB_ZH': 'data/train_vocab_zh.txt', 'RESULT_DIR': 'results/', 'LOG_DIR': 'logs/', 'CHK_DIR': 'checkpoints/', 'DATA_DIR': 'data/', 'MAX_INPUT_LENGTH': 40, 'MAX_VID_LENGTH': 32, 'batch_size': 512, 'wordembed_dim': 512, 'enc_hid_size': 512, 'dec_hid_size': 1024, 'teacher_ratio': 1, 'decoder_lr': 0.001, 'encoder_lr': 0.001, 'weight_decay': 5e-05, 'grad_clip': 10.0, 'epochs': 20, 'seed': 123, 'model_type': 's2s', 'train_id': 'en2zh'}
Vocab size src/tgt:10523/2907
train/val/test size: 254/30/59
#paras of my model: enc 12.220416M  dec 23.443803M total 35.664219M
Total time used: 0:04:26 Epoch 0 time uesd: 0:04:26 train loss: 4.1621 val loss: 7.2237 sentbleu: 0.2464 corpbleu: 0.1059
Finished 1 epochs of training
Total time used: 0:08:54 Epoch 1 time uesd: 0:04:27 train loss: 2.9013 val loss: 7.2622 sentbleu: 0.2652 corpbleu: 0.1407
Finished 2 epochs of training
Total time used: 0:13:22 Epoch 2 time uesd: 0:04:27 train loss: 2.5188 val loss: 7.2917 sentbleu: 0.2732 corpbleu: 0.1532
Finished 3 epochs of training
Total time used: 0:17:49 Epoch 3 time uesd: 0:04:27 train loss: 2.3404 val loss: 7.1805 sentbleu: 0.2850 corpbleu: 0.1700
Finished 4 epochs of training
Total time used: 0:22:18 Epoch 4 time uesd: 0:04:28 train loss: 2.2328 val loss: 7.2046 sentbleu: 0.2904 corpbleu: 0.1778
Finished 5 epochs of training
Total time used: 0:26:46 Epoch 5 time uesd: 0:04:27 train loss: 2.1572 val loss: 7.0789 sentbleu: 0.2984 corpbleu: 0.1902
Finished 6 epochs of training
Total time used: 0:31:15 Epoch 6 time uesd: 0:04:27 train loss: 2.0957 val loss: 7.1647 sentbleu: 0.3019 corpbleu: 0.1919
Finished 7 epochs of training
Total time used: 0:35:43 Epoch 7 time uesd: 0:04:27 train loss: 2.0431 val loss: 7.1279 sentbleu: 0.3061 corpbleu: 0.2015
Finished 8 epochs of training
Total time used: 0:40:11 Epoch 8 time uesd: 0:04:27 train loss: 1.9937 val loss: 7.1256 sentbleu: 0.3116 corpbleu: 0.2063
Finished 9 epochs of training
Total time used: 0:44:39 Epoch 9 time uesd: 0:04:27 train loss: 1.9505 val loss: 7.1055 sentbleu: 0.3146 corpbleu: 0.2124
Finished 10 epochs of training
Total time used: 0:49:07 Epoch 10 time uesd: 0:04:27 train loss: 1.9111 val loss: 7.0551 sentbleu: 0.3170 corpbleu: 0.2151
Finished 11 epochs of training
Total time used: 0:53:36 Epoch 11 time uesd: 0:04:28 train loss: 1.8742 val loss: 7.0822 sentbleu: 0.3188 corpbleu: 0.2184
Finished 12 epochs of training
Total time used: 0:58:04 Epoch 12 time uesd: 0:04:27 train loss: 1.8399 val loss: 7.0459 sentbleu: 0.3268 corpbleu: 0.2258
Finished 13 epochs of training
Total time used: 1:02:32 Epoch 13 time uesd: 0:04:27 train loss: 1.8076 val loss: 7.0140 sentbleu: 0.3327 corpbleu: 0.2347
Finished 14 epochs of training
Total time used: 1:07:00 Epoch 14 time uesd: 0:04:27 train loss: 1.7748 val loss: 6.9712 sentbleu: 0.3365 corpbleu: 0.2400
Finished 15 epochs of training
Total time used: 1:11:28 Epoch 15 time uesd: 0:04:27 train loss: 1.7442 val loss: 6.9522 sentbleu: 0.3432 corpbleu: 0.2485
Finished 16 epochs of training
Total time used: 1:15:56 Epoch 16 time uesd: 0:04:27 train loss: 1.7155 val loss: 6.9562 sentbleu: 0.3483 corpbleu: 0.2558
Finished 17 epochs of training
Total time used: 1:20:24 Epoch 17 time uesd: 0:04:27 train loss: 1.6913 val loss: 7.0026 sentbleu: 0.3518 corpbleu: 0.2602
Finished 18 epochs of training
Total time used: 1:24:53 Epoch 18 time uesd: 0:04:28 train loss: 1.6685 val loss: 6.9223 sentbleu: 0.3556 corpbleu: 0.2639
Finished 19 epochs of training
Total time used: 1:29:21 Epoch 19 time uesd: 0:04:27 train loss: 1.6488 val loss: 6.9468 sentbleu: 0.3618 corpbleu: 0.2718
Finished 20 epochs of training
Best corpus bleu score 0.2718 at epoch 19
************ Start eval... ************
Use epoch 19 as the best model for testing
30000

**Comment by zzxslp** [comment]
Hi bozhenhhu,
You are training with only 20 epochs and the model is not fully converged yet. Please use default hyper-parameters in config.yaml to reproduce the baseline results. Also the 0.15 you are referring to is BLEU-4 not corpus BLEU, you submission is ~0.28 and our baseline is 0.31. It is close, the only difference should be you modified the training epochs.
Please re-download our repo to reproduce the baseline if you made any changes to the code already.

Thanks

**Comment by PipiZong** [comment]
> Hi bozhenhhu,
> You are training with only 20 epochs and the model is not fully converged yet. Please use default hyper-parameters in config.yaml to reproduce the baseline results. Also the 0.15 you are referring to is BLEU-4 not corpus BLEU, you submission is ~0.28 and our baseline is 0.31. It is close, the only difference should be you modified the training epochs.
> Please re-download our repo to reproduce the baseline if you made any changes to the code already.
> 
> Thanks

Hi zzxslp:

Just wondering what's the difference between corpus bleu and bleu-4 here? And the BLEU_4 you mentioned in your paper stands for 'corpus bleu' instead of the one computed by the COCO standard evaluation code?
